test/index.spec.js

```diff
@@ -2,21 +2,35 @@
 /* eslint max-nested-callbacks: ["error", 8] */
 'use strict'
 
-const chai = require('chai')
-const dirtyChai = require('dirty-chai')
-const expect = chai.expect
-chai.use(dirtyChai)
+const { expect } = require('aegir/utils/chai')
 const multihash = require('multihashes')
 const multihashing = require('multihashing-async')
-const { Buffer } = require('buffer')
+const { utf8Encoder } = require('./helpers/encoder')
 
 const CID = require('../src')
+const toHexString = (buf) => {
+  return Array.from(buf).map(i => i.toString('16').padStart(2, '0')).join('')
+}
+
+const fromHexString = (str) => {
+  if (str.length % 2 !== 0) {
+    throw new Error('Invalid hex string')
+  }
+
+  const parts = []
+
+  for (let i = 0; i < str.length; i += 2) {
+    parts.push(parseInt(`${str[i]}${str[i + 1]}`, 16))
+  }
+
+  return Uint8Array.from(parts)
+}
```

Comment:
📝 I remember having to do this kind of thing elsewhere, maybe it's worth extracting it into HexEncoder / HexDecoder somewhere for sharing.

Reply:
Done!  Well, sort of - the `toString` and `fromString` functions take `base16` as an encoding.